### DISCUSSION TITLE
IE11

### DISCUSSION THREAD

**Discussion by lfbergee** [fremtind/jokul#1694] [Annet]
Den evige diskusjon, hva skal vi støtte, hva skal vi droppe.

Andre store aktører, inkludert Microsoft, dropper support på Teams og Office365, det burde sette i gang et nytt push for å bli kvitt IE for godt denne gangen.

Flere av våre dependecies har eller er iferd med å ta i bruk funksjonalitet som ikke fungerer, som gjør at vi potensielt må gjøre tiltak for å aktivt støtte IE, samtidig som det hindrer oss å ta ibruk funksjonalitet som feks Proxy. React testing libary bruker nettopp Proxy, men tilbyr en egen IE11 utgave for now. 

Vi har eksplisitt lagt til IE11 i vår browserslist. Dermed pakker vi alle våre pakker, og klientkode, for IE for ALLE. Det gjør at vi leverer en større bundle med flere polyfills enn vi hadde hatt behov for å gjøre. Ganske mange av våre CSS properties er duplisert for å ikke falle sammen i IE, jeg tipper vi kan shave ~20% av vår css ved å kvitte oss med det.

**Comment by espkva** [comment]
Microsoft tilbyr funksjonalitet for automatisk videresending av brukere fra IE til Edge hvis nettstedet er registrert hos Microsoft som inkompatibelt med IE.

https://docs.microsoft.com/en-us/microsoft-edge/web-platform/ie-to-microsoft-edge-redirection

Ser at Finn, VG, Tek.no blant andre har tatt det i bruk. Hele listen er tilgjengelig her: https://edge.microsoft.com/neededge/v1

@gbrownlee @lfbergee

**Reply by vegardsoerlie** [reply]
SeOpp er positive til en IE -> Edge redirect i hvert fall 👍

**Reply by lfbergee** [reply]
Som en utvikler som har brukt store deler av dagen på å 
1. få IE11-vmen til å virke
2. løse en super derpy feil ved at vi ikke transpilerte en tredjepart til es5
er jeg veldig veldig veldig for dette.

**Comment by gbrownlee** [comment]
Denne diskusjonen tar vi videre internt ifm. neste frontend-forum. Der lager vi en action-liste frem mot å sette en EOL-dato for IE i våre løsninger.

**Reply by wkillerud** [reply]
@gbrownlee begynner å bli lenge siden, men husker du hva IE11-diskusjonen landet på? Det ble relevant igjen i #2374

**Reply by gbrownlee** [reply]
Diskusjonen i forumet ble ikke ferdig, men både BM og ...et annet team... har avsluttet støtte for IE. Tror tanken var å benytte deres vurderinger som et rammeverk for andre løsninger. Hva husker du, @joms ?

**Reply by vegardsoerlie** [reply]
Mener at SeOpp også slutta med IE ;)

**Reply by joms** [reply]
@kennidenni har ikke dere klipt navlestrengen til IE11? Husker ikke helt hva vi kom fram til selv, men vi må ha en plan for å få alle konsumenter av Jøkul til å droppe støtte for IE.
Selv om man fortsatt ser mange brukere som bruker IE 11, må vi ta en del av samfunnsansvaret til å lede folk vekk fra IE 11. Her kan vi også trygt følge [SpareBank 1](https://www.kode24.no/kodenytt/derfor-kutter-sparebank-1-stotte-for-internet-explorer/73160899) sitt eksempel, som jeg vil si har tatt et ganske stort steg.

**Reply by kennidenni** [reply]
Jo stemmer, vi har kuttet det både i kundeklienten og for Smartforhandler (som bilforhandlere bruker) :) 
Vi valgte også å kutte gammel edge (som ikke er basert på chromium) da vi fant noen problemer med den også som vi ikke fikk testet på noen god måte.
Helt enig med @joms at vi burde droppe støtten og heller pushe brukerene våre til å bruke andre nettlesere. Vi på Motor kunne ikke se noen forskjell i antall brukere fordi de fleste hadde en annen nettleser installert fra før av. Vi fikk heller ingen tilbakemeldinger fra forhandlere om at det var dumt at vi gjorde det. 

@wkillerud Vi kunne kanskje vurdert å legge inn kode i Jøkul slik at det blir lettere for andre team å kopiere den løsningen vi har? Det vil jo ikke være mulig å ha det som en komponent da appen ikke mountes dersom du har IE, men det å ha koden klar for Copy-Paste kan være fint

**Reply by wkillerud** [reply]
Høres ut som en god bloggpost @kennidenni 😄

**Reply by kennidenni** [reply]
Kan se om jeg kan skrive det i neste uke!
